### ISSUE TITLE
nested namespaces on same not highlighted correctly

### ISSUE THREAD

**Issue by matter123** [jeff-hykin/cpp-textmate-grammar#132]
Code:
```c++
namespace foo { namespace bar {
	// foo
}} // namespace foo::bar
```
Image:
<img width="323" alt="Screen Shot 2019-04-26 at 14 32 40" src="https://user-images.githubusercontent.com/714007/56837777-75779180-6830-11e9-80e4-67a06f3bd45b.png">

`namespace bar` has the scope `meta.head.namespace.cpp`

**Comment by jeff-hykin** [comment]
<img width="405" alt="Screen Shot 2019-04-26 at 4 41 58 PM" src="https://user-images.githubusercontent.com/17692058/56838032-42d69480-6842-11e9-8b46-1936a4ea6ccb.png">
Looks like scoped namespaces also got messed up.

**Comment by matter123** [comment]
@jeff-hykin This is caused by the commit https://github.com/jeff-hykin/cpp-textmate-grammar/commit/4969c670b2cc0d8c2f680063fe455964d251510a. In the function BlockFinderFor on line 57, the start_pattern for the head range was changed from `/\G/` to `/\G| /`.

This captures a space if present after `{`, causing the lookbehind in the body to fail. Do you know why this change was made?

**Comment by jeff-hykin** [comment]
I know it was made to accommodate  something, I'll try to find it.

**Comment by matter123** [comment]
would maybe `/\G ?/` be correct? that is head optionally captures a space initially.

**Comment by jeff-hykin** [comment]
Yup that seems to work perfect 👌

**Comment by jeff-hykin** [comment]
<img width="315" alt="Screen Shot 2019-05-04 at 4 40 18 PM" src="https://user-images.githubusercontent.com/17692058/57185118-52318f80-6e8b-11e9-8db2-09351a2edc91.png">
Well except for this case

**Comment by jeff-hykin** [comment]
I think thats what the original change was made for.

**Comment by matter123** [comment]
That should probably be a separate release so hopefully, it can still be in Aprils release.

**Comment by jeff-hykin** [comment]
Yeah the switch issue is definitely preferable since it is not a big deal anyways.
It seems like `\G ?` should work if `\G| ` works. Maybe I don't fully understand the behavior of `\G`

**Comment by matter123** [comment]
`:switch_conditional_parentheses` needs to be removed from primary_includes

**Comment by matter123** [comment]
So `\G` is an anchor that matches the end of the start pattern. It is useful when you want a pattern to match only one time. like the head pattern on one of the options in `numeric_constant`.
in `switch(next) {` with `:switch_conditional_parentheses` in primary_includes, the `(` was matched outside of head. Once the `(` was matched `\G` could no longer match.

**Comment by matter123** [comment]
Alternatively, primary_includes could be moved after the head range.

**Comment by jeff-hykin** [comment]
Removing `switch_conditional_parentheses` from the head range fixed everything. 👍

**Comment by matter123** [comment]
`/ ?\G/`, is identical to `/\G/` because if the space ever matched \G couldn't

**Comment by jeff-hykin** [comment]
Oh so `\G` is the end of the previous regex match? If it was the end of the current match then wouldn't ` ?\G` work since if it matched the space, then G would be at the end of the space (the end of the start pattern)

**Comment by matter123** [comment]
\G is an anchor for the end of the `start_pattern` of the enclosing Range

**Comment by matter123** [comment]
Actually, yeah it looks like \G should also anchor to the end of the previous match.

**Comment by matter123** [comment]
imagine:

includes: [
/aa/,
/\Gbb/
]

that includes should match `aa` and `aabb` completely but in `aa bb` `bb` will not be matched as directly following the last match is not `bb`

**Comment by matter123** [comment]
This comment suggests that \G only works in a begin/end or begin/while rule.

https://github.com/Microsoft/vscode/issues/57831#issuecomment-420666621.

The above example does not work.

**Comment by matter123** [comment]
Fixed in v1.8.9.
